When doing `Put` for dotenv provider, an error is returned if a target file doesn't exist

Hi,

Very cool project! I would be interested to use it as Go API for accessing different secret storage backends (Delegated storage feature for https://github.com/capactio/capact open source project), as you prepared great abstractions.

Anyway, while playing around with Teller Go API, I noticed the following behavior for `dotenv` Provider. When doing `Put`, it returns an error when a given file doesn't exist.

I would love to prepare a pull request to fix such issue for `dotenv` provider. In fact, I have the code ready, but I want to confirm first the expected behavior.

## Expected behavior

I would expect that the `Put` method is doing upsert (create or update) operation, like `PUT` in REST API. By reading the code, this is how I believe the `vault`, `heroku` and `etcd` providers work.

## Minimal example to reproduce

```go
package main

import (
	tellerpkg "github.com/spectralops/teller/pkg"
	"github.com/spectralops/teller/pkg/core"
	"log"
)

func main() {
	providers := tellerpkg.BuiltinProviders{}
	provider, err := providers.GetProvider("dotenv")
	if err != nil {
		panic(err)
	}

	key := core.KeyPath{
		Path:  "/tmp/foo.env",
		Field: "data",
	}

	val := `{"key": "true"}`
	err = provider.Put(key, val)
	if err != nil {
		panic(err) // it will panic here
	}

	// Get
	entry, err := provider.Get(key)
	if err != nil {
		panic(err)
	}

	log.Printf("%+v\n", entry)
}
```

Result:

```
panic: open /tmp/foo.env: no such file or directory
```

--- a/pkg/providers/dotenv.go
+++ b/pkg/providers/dotenv.go
@@ -11,6 +11,11 @@ import (
 	"github.com/spectralops/teller/pkg/utils"
 )
 
+const (
+	filePerm = 0644
+	dirPerm  = 0755
+)
+
 type DotEnvClient interface {
 	Read(p string) (map[string]string, error)
 	Write(p string, kvs map[string]string) error
@@ -31,7 +36,7 @@ func (d *DotEnvReader) Write(p string, kvs map[string]string) error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(p, []byte(content), 0644)
+	return os.WriteFile(p, []byte(content), filePerm)
 }
 
 type Dotenv struct {
@@ -64,7 +69,8 @@ func (a *Dotenv) PutMapping(kp core.KeyPath, m map[string]string) error {
 	switch {
 	case err == nil:
 		// get a fresh copy of a hash
-		into, err := a.client.Read(p)
+		var into map[string]string
+		into, err = a.client.Read(p)
 		if err != nil {
 			return err
 		}
@@ -72,7 +78,7 @@ func (a *Dotenv) PutMapping(kp core.KeyPath, m map[string]string) error {
 		return a.client.Write(p, into)
 	case os.IsNotExist(err):
 		// ensure all subdirectories exist
-		err := os.MkdirAll(path.Dir(p), 0755)
+		err = os.MkdirAll(path.Dir(p), dirPerm)
 		if err != nil {
 			return err
 		}

--- a/pkg/providers/dotenv.go
+++ b/pkg/providers/dotenv.go
@@ -2,6 +2,7 @@ package providers
 
 import (
 	"os"
+	"path"
 	"sort"
 
 	"github.com/joho/godotenv"
@@ -58,13 +59,27 @@ func (a *Dotenv) PutMapping(kp core.KeyPath, m map[string]string) error {
 		return err
 	}
 
-	// get a fresh copy of a hash
-	into, err := a.client.Read(p)
-	if err != nil {
+	// check if the file does exist
+	_, err = os.Stat(p)
+	switch {
+	case err == nil:
+		// get a fresh copy of a hash
+		into, err := a.client.Read(p)
+		if err != nil {
+			return err
+		}
+		utils.Merge(m, into)
+		return a.client.Write(p, into)
+	case os.IsNotExist(err):
+		// ensure all subdirectories exist
+		err := os.MkdirAll(path.Dir(p), 0755)
+		if err != nil {
+			return err
+		}
+		return a.client.Write(p, m)
+	default:
 		return err
 	}
-	utils.Merge(m, into)
-	return a.client.Write(p, into)
 }
 
 func (a *Dotenv) GetMapping(p core.KeyPath) ([]core.EnvEntry, error) {

